server/src/test/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregatorTests.java

```diff
@@ -42,1236 +42,673 @@
-                        assertEquals(startDate.plusHours(i), bucket.getKey());
-                        assertEquals(1, bucket.getDocCount());
-                    }
-                });
-        testSearchAndReduceCase(new MatchAllDocsQuery(), dataset,
-                aggregation -> aggregation.setNumBuckets(300).field(DATE_FIELD),
-                histogram -> {
-                    List<? extends Histogram.Bucket> buckets = histogram.getBuckets();
-                    assertEquals(200, buckets.size());
-                    for (int i = 0; i < 200; i++) {
-                        Histogram.Bucket bucket = buckets.get(i);
-                        assertEquals(startDate.plusHours(i * 3), bucket.getKey());
-                        assertEquals(3, bucket.getDocCount());
-                    }
-                });
-        testSearchAndReduceCase(new MatchAllDocsQuery(), dataset,
-                aggregation -> aggregation.setNumBuckets(100).field(DATE_FIELD),
-                histogram -> {
-                    List<? extends Histogram.Bucket> buckets = histogram.getBuckets();
-                    assertEquals(50, buckets.size());
-                    for (int i = 0; i < 50; i++) {
-                        Histogram.Bucket bucket = buckets.get(i);
-                        assertEquals(startDate.plusHours(i * 12), bucket.getKey());
-                        assertEquals(12, bucket.getDocCount());
-                    }
-                });
-        testSearchAndReduceCase(new MatchAllDocsQuery(), dataset,
-                aggregation -> aggregation.setNumBuckets(30).field(DATE_FIELD),
-                histogram -> {
-                    List<? extends Histogram.Bucket> buckets = histogram.getBuckets();
-                    assertEquals(25, buckets.size());
-                    for (int i = 0; i < 25; i++) {
-                        Histogram.Bucket bucket = buckets.get(i);
-                        assertEquals(startDate.plusDays(i), bucket.getKey());
-                        assertEquals(24, bucket.getDocCount());
-                    }
-                });
+        final Map<Integer, Integer> bucketsToExpectedDocCountMap = new HashMap<>();
+        bucketsToExpectedDocCountMap.put(seed, 1);
+        bucketsToExpectedDocCountMap.put(seed / 2, 3);
+        bucketsToExpectedDocCountMap.put(seed / 6, 12);
+        bucketsToExpectedDocCountMap.put(seed / 20, 24);
+        final Integer randomNumberOfBuckets = randomFrom(bucketsToExpectedDocCountMap.keySet());
+        testSearchAndReduceCase(DEFAULT_QUERY, dataset,
+            aggregation -> aggregation.setNumBuckets(randomNumberOfBuckets).field(DATE_FIELD),
+            histogram -> {
+                final List<? extends Histogram.Bucket> buckets = histogram.getBuckets();
+                final int expectedDocCount = bucketsToExpectedDocCountMap.get(randomNumberOfBuckets);
+                final int expectedSizeCeil = (seed + expectedDoc
```

Comment:
Yes, 100000 was really generous. I tested with some values like 2500, which made the tests fail as it would only allow 83 buckets or so for the tests. Hence 25000 seemed to be a good approximation for all test cases

Reply:
Thanks, makes sense.